lib/cli/nodesetupcommand.cpp

```diff
@@ -530,20 +525,14 @@ int NodeSetupCommand::SetupNode(const boost::program_options::variables_map& vm,
 	if (!ticket.IsEmpty()) {
 		String ticketPath = ApiListener::GetCertsDir() + "/ticket";
 
-		String tempTicketPath = Utility::CreateTempFile(ticketPath + ".XXXXXX", 0600, fp);
+		AtomicFile::Write(ticketPath, 0600, ticket);
 
-		if (!Utility::SetFileOwnership(tempTicketPath, user, group)) {
+		if (!Utility::SetFileOwnership(ticketPath, user, group)) {
```

Comment:
This can still be done on the file while it still has its temporary name. Either expose the temporary name or implement a matching function on `AtomicFile`.

Reply:
What for?

Reply:
Well, given that it just continues with just a log message if the `chmod()` fails, this doesn't seem to be too important. However, in general, it would be nice if the file had the correct attributes when given its final name.

I haven't checked if this can happen here, but if this file is to be picked up by the daemon while this command is running, there would be a brief period where it would find the file but fails to open it due to wrong permissions, which shouldn't happen.

Reply:
That command is run exactly once, then one restarts Icinga 2. Also:

> given that it just continues with just a log message if the `chmod()` fails, this doesn't seem to be too important

Reply:
Adding `AtomicFile::GetTempName()` would be a tiny addition and this chown call could just stay like it was before (which is the nicer order of operations IMHO).